src/include/daos_metrics.h

```diff
@@ -0,0 +1,268 @@
+	daos_rpc_cntr_t crc_attr_cnt;
+	/** Counter for container query rpc calls */
+	daos_rpc_cntr_t crc_query_cnt;
+	/** Counter for container oidalloc rpc calls */
+	daos_rpc_cntr_t crc_oidalloc_cnt;
+	/** Counter for container aggregate rpc calls */
+	daos_rpc_cntr_t crc_aggregate_cnt;
+} daos_cont_rpc_cntrs_t;
+
+/** rpc counters associated with DAOS Objects  */
+typedef struct {
+	/** Counter for all object related rpc calls */
+	daos_rpc_cntr_t orc_op_cnt;
+	/** Counter for object update rpc calls */
+	daos_rpc_cntr_t orc_update_cnt;
+	/** Counter for object fetch rpc calls */
+	daos_rpc_cntr_t orc_fetch_cnt;
+	/** Counter for object punch rpc calls */
+	daos_rpc_cntr_t orc_obj_punch_cnt;
+	/** Counter for dkey punch rpc calls */
+	daos_rpc_cntr_t orc_dkey_punch_cnt;
+	/** Counter for akey punch rpc calls */
+	daos_rpc_cntr_t orc_akey_punch_cnt;
+	/** Counter for object list rpc calls */
+	daos_rpc_cntr_t orc_obj_enum_cnt;
+	/** Counter for dkey enumerate rpc calls */
+	daos_rpc_cntr_t orc_dkey_enum_cnt;
+	/** Counter for akey enumerate rpc calls */
+	daos_rpc_cntr_t orc_akey_enum_cnt;
+	/** Counter for recx enumerate rpc calls */
+	daos_rpc_cntr_t orc_recx_enum_cnt;
+	/** Counter for obj sync rpc calls */
+	daos_rpc_cntr_t orc_sync_cnt;
+	/** Counter for obj query key rpc calls */
+	daos_rpc_cntr_t orc_querykey_cnt;
+	/** Counter for obj compound(tx) rpc calls */
+	daos_rpc_cntr_t orc_cpd_cnt;
+} daos_obj_rpc_cntrs_t;
+
+/** rpc counters for Pool, Container and Object calls */
+typedef struct {
+	daos_pool_rpc_cntrs_t arc_pool_cntrs;
+	daos_cont_rpc_cntrs_t arc_cont_cntrs;
+	daos_obj_rpc_cntrs_t  arc_obj_cntrs;
+} daos_all_rpc_cntrs_t;
+
+/** Metric type for stats */
+typedef struct {
+	unsigned long st_value;
+	unsigned long st_min;
+	unsigned long st_max;
+	unsigned long st_sum;
+	unsigned long st_sum_of_squares;
+} daos_stats_t;
+
+/** daos i/o stats */
+typedef struct {
+	daos_stats_t ios_obj_update;
+	daos_stats_t ios_obj_fetch;
+} daos_iostats_t;
+
+/** Types of i/o stats */
+enum daos_io_type {
+	DM_OBJ_UPDATE	= 0x001,
+	DM_OBJ_FETCH	= 0x002,
+};
+
+/** Metrics for i/o distribution */
+/**
+ * Distribution ids for fetch/update rpc calls based on size.
+ * Field DM_IO_X_Y indicates calls with size greater than X and less than
+ * or equal to Y
+ */
+enum {
+	DM_IO_0_1K = 0,
+	DM_IO_1K_2K = 1,
+	DM_IO_2K_4K,
+	DM_IO_4K_8K,
+	DM_IO_8K_16K,
+	DM_IO_16K_32K,
+	DM_IO_32K_64K,
+	DM_IO_64K_128K,
+	DM_IO_128K_256K,
+	DM_IO_256K_512K,
+	DM_IO_512K_1M,
+	DM_IO_1M_2M,
+	DM_IO_2M_4M,
+	DM_IO_4M_INF,
+	DM_IO_BYSIZE_COUNT,
+};
+
+/** Distribution of i/o rpc calls based on size */
+typedef struct {
+	unsigned long	idp_total_call_cnt;
+	unsigned long	idp_buckets[DM_IO_BYSIZE_COUNT];
+} daos_iocall_dist_byprot_t;
+
+/** Distribution ids for update rpc calls based on protection/replication type */
+enum {
+	/** No Protection */
+	DM_IO_NO_PROT = 0,
+	/** Replication */
+	DM_IO_RP,
+	/** Erasure coding and partial update of a stripe */
+	DM_IO_EC_P,
```

Comment:
you mentioned you were going to add some explicit counters for common redundancy classes?